nibabel/testing/__init__.py

```diff
@@ -62,6 +65,22 @@ def assert_allclose_safely(a, b, match_nans=True, rtol=1e-5, atol=1e-8):
     assert_true(np.allclose(a, b, rtol=rtol, atol=atol))
 
 
+def check_iteration(iterable):
```

Comment:
I was doing like you said before but I discover that `list(iterable)` calls `len` on `iterable` first. That was interfering with my tests for the `LazyTractogram` that is why I'm doing an explicit for loop.